### PR TITLE
Enable ZMQ timestamps and updated XLA cost model dataset generation

### PR DESCRIPTION
**Pull request by chenyu-jiang**

### CODE REVIEW

File: optimizer.py [hunk 1]
```diff
@@ -267,12 +267,13 @@ def op_fusion(self, _dag, _pkg: PKGraph, u_, v_):
...
-            _pkg.contract_edge(u0_, v0_)
-            self._fuse_pair(_dag, u0_, v0_)
+            # u0_ = self._debug_convert_to_the_other_machine(u_)
+            # v0_ = self._debug_convert_to_the_other_machine(v_)
+            # _pkg.contract_edge(u0_, v0_)
+            # self._fuse_pair(_dag, u0_, v0_)
             # self._fuse_pair(_dag, self.convert_fw2bw(v_), self.convert_fw2bw(u_))
-            return True, set([u_+"+"+v_, u0_+"+"+v0_]), set([u_, v_, u0_, v0_])
+            # return True, set([u_+"+"+v_, u0_+"+"+v0_]), set([u_, v_, u0_, v0_])
+            return True, set([u_+"+"+v_]), set([u_, v_])
```
Comment (joapolarbear) [inline optimizer.py]: Why are these lines commented out? What I understand is these lines are used to accelerate searching.

Comment (chenyu-jiang) [reply optimizer.py]: These lines were trying to replicate strategies in one node to others, but only works when there are only two workers. I think a more general solution (that also apply to other optimizations) is needed.

File: replay.py [hunk 2]
```diff
@@ -411,7 +411,24 @@ def reset_replayer(self):
...
+		# 		prev_edge_attrs = self.dag[node][succ]
+		# 		self.dag.remove_edge(node, succ)
+		# 		merge_node_name = "{}_MERGE{}".format(node, merge_counter)
+		# 		merge_counter += 1
+		# 		self.dag.add_node(merge_node_name, avg=0.005, GAP_STR_OP2OP=1, GAP_STR_OP2COMM=1)
+		# 		self.dag.add_edge(node, merge_node_name)
+		# 		for key, val in prev_edge_attrs.items():
+		# 			self.dag[node][merge_node_name][key] = val
+		# 		self.dag.add_edge(merge_node_name, succ, avg=0.05, GAP_STR_OP2OP=1, GAP_STR_OP2COMM=1)
+
```
Comment (joapolarbear) [inline replay.py]: Is self.orig_dag necessary?  Is that used in other places? If not, we can only use self.dag. And remove these comments if they are not necessary.

Comment (chenyu-jiang) [reply replay.py]: Yes I think we can remove these lines and self.orig_dag. They are originally for adding a "merge" node into the graph after each XLA cluster, but later I found that those nodes are just to implement branching and do not corresponding to actual GPU computation, so we do not need to add those nodes.

**Comment by chenyu-jiang** [comment]
Conflicts are resolved but not tested yet.

**Pull request merged**